### PR TITLE
[docs] updated README for OSX install

### PR DESCRIPTION
**Pull request by thierrygosselin**

### CODE REVIEW

**Comment by StrikerRUS** [comment]
I can't review this PR since I have no Mac, but there are the same comments at 46 line and at 37, 72 lines in [Python-package file](https://github.com/Microsoft/LightGBM/blob/master/python-package/README.rst).

File: R-package/README.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ Install LightGBM R-package with the following command:
 ```sh
 git clone --recursive https://github.com/Microsoft/LightGBM
 cd LightGBM/R-package
-# export CXX=g++-7 CC=gcc-7 # for OSX
+#  export CC=/usr/local/bin/gcc CXX=/usr/local/bin/g++ CXX1X=/usr/local/bin/g++ # for OSX
```
Comment (henry0312) [inline R-package/README.md]: if you install gcc via Homebrew, `/usr/local/bin/gcc` and `/usr/local/bin/g++` don't exist.

```
~
❯ ls /usr/local/bin/gcc
ls: /usr/local/bin/gcc: No such file or directory

~
❯ ls /usr/local/bin/gcc-*
/usr/local/bin/gcc-6@        /usr/local/bin/gcc-ar-6@     /usr/local/bin/gcc-nm-6@     /usr/local/bin/gcc-ranlib-6@
/usr/local/bin/gcc-7@        /usr/local/bin/gcc-ar-7@     /usr/local/bin/gcc-nm-7@     /usr/local/bin/gcc-ranlib-7@
```

**Comment by guolinke** [comment]
@thierrygosselin close ?

**Comment by thierrygosselin** [comment]
Yes, sorry I think I'm going to remain a passive user.
I really don't have the time to argue for days two dots here and there in doc.
Worked for me on 3 macs.

I lost so many hours with Macport and Homebrew with genomic workflows
that I've become allergic to these shortcuts.
Anyway, seems to work fine for people to use Homebrew and LightGBM
Best
Thierry

**Pull request closed without merging**